e2e/nomostest/ntopts/multi_repo.go

```diff
@@ -59,12 +57,35 @@ type MultiRepo struct {
 	RepoSyncPermissions []rbacv1.PolicyRule
 }
 
+// GitSourceOption mutates a GitSyncSource
+type GitSourceOption func(*syncsource.GitSyncSource)
+
+// Unstructured mutates a GitSyncSource to use the unstructured SourceFormat.
+func Unstructured(source *syncsource.GitSyncSource) {
+	source.SourceFormat = configsync.SourceFormatUnstructured
+}
+
 // SyncWithGitSource tells the test case that a RootSync or RepoSync should be
 // applied that points to an empty Git Repository.
 // TODO: Add another option that allows specifying an existing Repository
-func SyncWithGitSource(id core.ID) func(opt *New) {
+func SyncWithGitSource(id core.ID, gitOpts ...GitSourceOption) func(opt *New) {
```

Comment:
Do we need the granularity of specifying source format for each git source? I could see it potentially being useful, but most of the time it seems a bit less user friendly to enumerate the option for every RSync used by the test